pkg/process/checks/net.go

```diff
@@ -126,12 +132,81 @@ func (c *ConnectionsCheck) Realtime() bool { return false }
 // ShouldSaveLastRun indicates if the output from the last run should be saved for use in flares
 func (c *ConnectionsCheck) ShouldSaveLastRun() bool { return false }
 
-// Run runs the ConnectionsCheck to collect the active network connections
-// and any closed network connections since the last Run.
-// For each connection we'll return a `model.Connection`
-// that will be bundled up into a `CollectorConnections`.
-// See agent.proto for the schema of the message and models.
-func (c *ConnectionsCheck) Run(nextGroupID func() int32, _ *RunOptions) (RunResult, error) {
+func (c *ConnectionsCheck) runGRPC(nextGroupID int32) (RunResult, error) {
+	// Create a context with a timeout of 10 seconds
+	timedContext, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	opts := []grpc.DialOption{
+		grpc.WithContextDialer(func(_ context.Context, _ string) (nnet.Conn, error) {
+			return nnet.Dial(api.NetType, c.syscfg.GRPCSocketFilePath)
```

Comment:
This should use the constant in `pkg/process/net` rather than importing `cmd/system-probe/api`, which shouldn't be used outside of system-probe.